### ISSUE TITLE
Update "The Architectural Structure of a UI Component" Link

### ISSUE THREAD

**Issue by ryantfowler** [magento/devdocs#1806]
### Description
* Potentially incorrect url or misleading URL link name
* The [following](http://devdocs.magento.com/guides/v2.2/ui_comp_guide/ui_comp_outline_proposed.html) link appears to be incorrect, as it goes to a page that refers to overall M2 architecture and is not specific to UI Component architecture ... so if it's intentionally supposed to go to this link, then the link name is misleading. If it isn't meant to go to this page, then I wonder if it's meant to go to another?

**Comment by bdenham** [comment]
Hey @ryantfowler, I've recently removed the page that your referenced link refers to because it was a proposed outline for a new UI Components guide, which I'm currently working on. Which page contained this link?

**Comment by ryantfowler** [comment]
@bdenham I'm not following your question.

 I originally opened this issue because I visited [this](https://devdocs.magento.com/guides/v2.2/ui_comp_guide/ui_comp_outline_proposed.html) page ... and on this page there is a link [The architectural structure of a UI component](https://devdocs.magento.com/guides/v2.2/architecture/archi_perspectives/components/AComponents.html) which takes a visitor to a page that has the title "Magento components" and has a list of what these components are: Modules, Themes, Language packages. Therefore, there is a link on the proposed UI Components Guide, which is refers to the architecture structure of a *UI Component*, but instead links to a page that discusses "Components" in the sense of what is within this file: `\Magento\Framework\Component\ComponentRegistrar` .

What I *would* expect, based on the title of the link (The architectural structure of a UI component), would be a link to a page or section of the documentation which describes the actual architectural structure of a `UI Component` which is related to any other part of the "UI Components Guide."

What I am *actually* seeing, is a link "The architectural structure of a UI component" which goes to an almost completely unrelated part of the Magento system, and certainly a different type of "component." I acknowledge that the term "component" is used very liberally in the core code, and throughout M2, so it can be confusing as to what someone is actually referring to when that term is used.

TLDR: There's a link on the "UI Components Guide" which leads to an incorrect page that is unrelated to the guide as it is in it's current state. Please request a technical review of what I'm saying for further validation.

Does this help? Thanks

**Comment by bdenham** [comment]
@ryantfowler, thanks for the explanation. 

I think the real issue is that you want to learn about the architecture of a UI component, but can't find any information about it. The link made a promise it couldn't fulfill because there is currently no published UI component architecture doc. We will change that as soon as we can by adding a new topic to the Concepts section of the UI Components guide as soon as we can.

In moving towards that goal, I just deleted the original page you visited with the offending link [UI Components Outline (proposed)](https://devdocs.magento.com/guides/v2.2/ui_comp_guide/ui_comp_outline_proposed.html) because it is no longer useful for inclusion within the guide. It was included long ago as a preview of what's to come in our revision of this guide. But that has since changed.

This root of this issue (missing Architecture Overview of a UI Component) is being tracked in Jira: https://jira.corp.magento.com/browse/MAGEDOC-2965

**Comment by ryantfowler** [comment]
@bdenham yes, that is correct :). Thank you! I'll close this now.
